sunpy/data/data_manager/cache.py

```diff
@@ -167,5 +167,6 @@ def download(url):
                 return download(url)
             except Exception as e:
                 warn(e, SunpyUserWarning)
+                error = e
```

Comment:
why do we need this line, is `e` really not still accessible after the except block?

Reply:
It is not unfortunately. I thought the same thing at first, but my editor complained about it being undefined in the scope of that last `else` block.

Reply:
Do we not want to append each error to a list and then raise the entire bunch of errors instead of the last error?

Reply:
I agree that it is ugly

Reply:
🤷 we could? I don't have a strong opinion on it. They each get raised as warnings along the way anyway.

Reply:
The disadvantage would be that you can't raise a `RuntimeError` from a list of exceptions.